gov.pnnl.goss.gridappsd/src/gov/pnnl/goss/gridappsd/data/conversion/ProvenWeatherToGridlabdWeatherConverter.java

```diff
@@ -0,0 +1,244 @@
+						GridAppsDConstants.topic_platformLog);
+			}
+		}
+	}
+	
+	
+	@Override
+	public void convert(String inputContent, PrintWriter outputContent) throws Exception {
+		boolean headerPrinted = false;
+		
+		TimeSeriesResult resultObj = TimeSeriesResult.parse(inputContent);
+		for(TimeSeriesMeasurementResult record: resultObj.getMeasurements()){
+			if(!headerPrinted){
+				printGLDHeader(record, outputContent);
+				headerPrinted = true;
+			}
+			convertRecord(record, outputContent);
+		}
+	}
+
+	@Override
+	public void convert(InputStream inputContent, PrintWriter outputContent)  throws Exception {
+		boolean headerPrinted = false;
+		
+		String strContent = IOUtils.toString(inputContent);
+		TimeSeriesResult resultObj = TimeSeriesResult.parse(strContent);
+		for(TimeSeriesMeasurementResult record: resultObj.getMeasurements()){
+			if(!headerPrinted){
+				printGLDHeader(record, outputContent);
+				headerPrinted = true;
+			}
+			convertRecord(record, outputContent);
+		}
+
+	}
+
+	protected void printGLDHeader(TimeSeriesMeasurementResult record, PrintWriter outputContent){
+		//TODO this needs to come from data or tags within proven
+		String placeStr="", yearStr="", latlong = "";
+		
+		try{
+		HashMap<String, String> map = record.getPoints().get(0).getRow().getEntryMap();
+		placeStr = map.get(PLACE);
+		placeStr = placeStr.replaceAll("\"", "");
+		latlong = map.get(LATITUDE)+","+map.get(LONGITUDE);
+		
+		String dateStr = map.get(DATE);
+		String[] dateArr = StringUtils.split(dateStr,"/");
+		yearStr = dateArr[2];
+		}catch (Exception e) {
+			//TODO log warning
+		}
+		
+		outputContent.println("#"+placeStr+" ("+latlong+") file for "+yearStr);
+		outputContent.println("# data obtained from ...");
+		outputContent.println("$state_name=N/A");
+		outputContent.println("$city_name=N/A");
+		
+		outputContent.println("temperature,humidity,wind_speed,solar_dir,solar_diff,solar_global");
+	}
+	
+	protected void convertRecord(TimeSeriesMeasurementResult record, PrintWriter outputContent){
+		//See https://github.com/gridlab-d/gridlab-d/blob/master/climate/climate.cpp for gridlabd format requirements
+		for(TimeSeriesRowResult result: record.getPoints()){
+			Map<String, String> map = result.getRow().getEntryMap();
+			
+			String dateStr = map.get(DATE);
+			String timeStr = map.get(MST);
+			try {
+				Date datetime = sdfIn.parse(dateStr+" "+timeStr);
+				outputContent.print(sdfOut.format(datetime)+",");
+			} catch (ParseException e) {
+				e.printStackTrace();
+				//todo throw exception
+			}
+			
+
+			double temp_fahr = readDouble(map, TEMPERATURE, -100000000);
+			//we are already receiving it as fahrenhight
+			//print temperature in Fahrenheit and convert from celcius to Fahrenheit
+//			double temp_cel = readDouble(map, TEMPERATURE, -100000000);
+//			double temp_fahr = (temp_cel * 1.8) + 32;
+			outputContent.print(temp_fahr+",");
+			//print humidity 
+			outputContent.print(map.get(HUMIDITY)+",");
```

Comment:
Make sure Humidity is per unit (0 < x < 1). as simple check to see if Humidity is greater than one then divide by 100.